src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketWebhookClient.java

```diff
@@ -0,0 +1,39 @@
+package com.atlassian.bitbucket.jenkins.internal.client;
+
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketPage;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketWebhook;
+import com.fasterxml.jackson.core.type.TypeReference;
+import okhttp3.HttpUrl;
+
+import static java.util.Arrays.stream;
+
+public class BitbucketWebhookClient {
```

Comment:
I'm not sold on this structure, it is different from how the client was meant to be (and indeed is)  built.
the idea would be that I can do 
`clientfactory.getProjectClient("projectKey").getRepositoryClient("repoSlug").getWebhooks(String ... eventIdFilter)`
*or* (if needed)
`clientfactory.getProjectClient("projectKey").getRepositoryClient("repoSlug").getWebhookClient().get(Strin eventIdFilter)`

The whole idea around the client was that it was fluent and typesafe.
I think this belongs on the `repositoryClient`

Reply:
I totally missed that. I agree.